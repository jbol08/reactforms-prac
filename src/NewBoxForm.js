import React, { useState } from 'react';
import { v4 as uuid } from "uuid";

const NewBoxForm = ({makeBox}) => {
    const [formData, setFormData] = useState({
        backgroundColor: '',
        height: '',
        width: ''
    });

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(formData => ({
          ...formData,
          [name]: value
        }));
      }
    const getValues = (evt) => {
        evt.preventDefault();
        makeBox({...formData, id: uuid()});
        setFormData({ backgroundColor: '', height: '', width: '' })

    };

    return (
        <div>
            <form onSubmit={getValues}>
                <div>
                    <label htmlFor='backgroundColor'>Background Color:</label>
                    <input onChange={handleChange}
                        name='backgroundColor'
                        type='text'
                        value={formData.backgroundColor}
                        id='backgroundColor' />
                </div>
                <div>
                    <label htmlFor='height'>Height:</label>
                    <input onChange={handleChange}
                        name='height'
                        type='text'
                        value={formData.height}
                        id='height' />
                </div>
                <div>
                    <label htmlFor='width'>Width:</label>
                    <input onChange={handleChange}
                        name='width'
                        type='text'
                        value={formData.width}
                        id='width' />
                </div>
                <button> Add a New Box</button>
            </form>
        </div>
    );

}

export default NewBoxForm;