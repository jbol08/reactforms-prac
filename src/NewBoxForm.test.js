import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";


//smoke test
it('renders the div without crashing', () => {
    render(<NewBoxForm />)
});

//snapshot test
it('takes a snapshot of the initial setup', () => {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot()
});
