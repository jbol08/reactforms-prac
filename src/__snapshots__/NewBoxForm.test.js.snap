// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`takes a snapshot of the initial setup 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <label
          for="backgroundColor"
        >
          Background Color:
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="height"
        >
          Height:
        </label>
        <input
          id="height"
          name="height"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="width"
        >
          Width:
        </label>
        <input
          id="width"
          name="width"
          type="text"
          value=""
        />
      </div>
      <button>
         Add a New Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
